src/test/java/school/lemon/changerequest/java/junit/hw1/task1/api/ContactApiTest.java

```diff
@@ -0,0 +1,215 @@
+        Contact afterUpdate = contactApi.get((long) 1002);
+        assertEquals(beforeUpdate, afterUpdate);
+        assertTrue(result);
+    }
+
+    @Test
+    public void updateNotExistedContactTest() {
+        boolean result = contactApi.update(testContact);
+        assertFalse(result);
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void updateNullTest() throws Exception {
+        contactApi.update(null);
+    }
+
+    @Test
+    public void createTest() {
+        assertEquals(3, contactApi.filledContactsAmount());
+        try {
+            contactApi.find(testContact.getCallNumber());
+            fail();
+        } catch (ContactNotFoundException e) {
+        }
+
+        contactApi.create(testContact);
+        assertEquals(4, contactApi.filledContactsAmount());
+
+        Contact result = contactApi.find(testContact.getCallNumber());
+        assertEquals(testContact, result);
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void createNullTest() {
+        contactApi.create(null);
+    }
+
+    @Test(expected = IllegalStateException.class)
+    public void createFullArrayTest() {
+        fillWithContacts();
+        Contact contact = new Contact((long) 0, new CallNumber(randomNumeric(2), randomNumeric(3), randomNumeric(8)), "Test", "Test");
+        contactApi.create(contact);
+    }
+
+    @Test
+    public void getAllTest() {
+        Contact[] list = contactApi.getAll();
+        initContactsTest(list);
+
+        contactApi.create(testContact);
+        list = contactApi.getAll();
+        initContactsTest(list);
+        assertEquals(testContact, list[3]);
+    }
+
+    @Test
+    public void getAllEmptyListTest() {
+        deleteAllInitContacts();
+        Contact[] emptyList = contactApi.getAll();
+        assertEquals(0, emptyList.length);
+    }
+
+    private void initContactsTest(Contact[] list) {
+        boolean firstRes = cmpWithContact(list[0], (long) 1000, "Iurii", "Sergiichuk");
+        assertTrue(firstRes);
+
+        boolean secondRes = cmpWithContact(list[1], (long) 1001, "Dmytro", "Mudrov");
+        assertTrue(secondRes);
+
+        boolean thirdRes = cmpWithContact(list[2], (long) 1002, "Test", "Test");
+        assertTrue(thirdRes);
+    }
+
+    @Test
+    public void filledContactsAmountTest() {
+        int amount = contactApi.filledContactsAmount();
+        assertEquals(3, amount);
+
+        contactApi.create(testContact);
+        amount = contactApi.filledContactsAmount();
+        assertEquals(4, amount);
+
+        contactApi.delete((long) 1003);
+        amount = contactApi.filledContactsAmount();
+        assertEquals(3, amount);
+
+        fillWithContacts();
+        amount = contactApi.filledContactsAmount();
+        assertEquals(ContactApi.CONTACT_API_ARRAY_SIZE, amount);
+    }
+
+    private void deleteAllInitContacts() {
+        if (contactApi != null) {
+            contactApi.delete((long) 1000);
```

Comment:
Too much `black magic`... Get rid of those `magic numbers`: extract them as constants, pass as arguments, etc.

This test right now could not be maintained.